### ISSUE TITLE
☑️ Deploy new date picker

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#6352]
This issue is for deploying the new date picker (#5645 ). It also has some overlap with #6930

For experience cards (where we create and edit experience), these date pickers should only have Year, Month.

Everywhere else, it should be Year-Month-Day. If you find a spot you think should be different, you can also ask a designer or Gray.

## Acceptance Criteria
- [ ] All existing date inputs are replaced with Date Picker Fieldset from #5645 
- [ ] Date inputs for experiences should only have Year, Month
- [ ] All other date pickers have Year, Month, Day. If you find a spot where you think it should be different, talk to a designer.
